disassemblers/ofrak_ghidra/ofrak_ghidra/components/blocks/unpackers.py

```diff
@@ -35,31 +35,40 @@ class GhidraCodeRegionUnpacker(CodeRegionUnpacker, OfrakGhidraMixin):
         os.path.join(CORE_OFRAK_GHIDRA_SCRIPTS, "GetComplexBlocks.java"),
     )
 
+    get_code_regions_script = OfrakGhidraScript(
+        os.path.join(CORE_OFRAK_GHIDRA_SCRIPTS, "GetCodeRegions.java"),
+    )
+
     async def unpack(self, resource: Resource, config=None) -> None:
         program = await resource.get_only_ancestor_as_view(
             GhidraProject, ResourceFilter(tags=[GhidraProject], include_self=True)
         )
         program_attributes = await program.resource.analyze(ProgramAttributes)
         code_region = await resource.view_as(CodeRegion)
-        code_region_start = code_region.virtual_address
+        code_region_start = await self.fixup_address(code_region.virtual_address, program.resource)
         code_region_end = code_region_start + code_region.size
+        va_addend = code_region.virtual_address - code_region_start
 
         complex_blocks = await self.get_complex_blocks_script.call_script(
             resource,
             hex(code_region_start),
             hex(code_region_end),
         )
-        complex_blocks_created = list()
+
+        complex_blocks_created = []
+
         for complex_block in complex_blocks:
             complex_block = ComplexBlock(
-                complex_block["loadAddress"], complex_block["size"], complex_block["name"]
+                complex_block["loadAddress"] + va_addend, complex_block["size"], complex_block["name"]
```

Comment:
This PR gets a lot simpler if you don't add the `va_addend` here and just let it use the vaddr the backend provides. Then the created ComplexBlocks will all have the "correct" vaddr and then we don't need to do any later checks when unpacking CBs or BBs.

Reply:
Wouldn't then ComplexBlocks (and then consequently BasicBlocks and Instructions) have virtual addresses that are either not contained or misaligned with the CodeRegion's virtual address? This seems not good to me. That's why I think we should switch to setting the CodeRegion's virtual address using the one from the backend.

Reply:
Yes it will be mismatched, that's why correcting for the misaligned code regions so we can get complex blocks was discussed as the first step towards feature completion. So now we can sync up on what is the best path forward.